library/java/org/chromium/net/impl/CronetBidirectionalState.java

```diff
@@ -0,0 +1,431 @@
+package org.chromium.net.impl;
+
+import androidx.annotation.IntDef;
+
+import org.chromium.net.RequestFinishedInfo;
+import org.chromium.net.impl.RequestFinishedInfoImpl.FinishedReason;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ *  Holder the the current state associated to a bidirectional stream. The main goal is to provide
+ *  a mean to determine what should be the next action for a given event by considering the
+ *  current state. This class uses Compare And Swap logic. The next state is saved with
+ *  {@code AtomicInteger.compareAndSet()}.
+ *
+ *  <p>All methods in this class are Thread Safe.
+ */
+final class CronetBidirectionalState {
+
+  /**
+   * Enum of the events altering the global state. There are 3 types of events: User induced
+   * (prefixed with USER_), EM Callbacks (prefixed with ON_), and internal events (the remaining
+   * ones).
```

Comment:
Your call, but given the description in this comment, I expected to see all the USER_ values first, then the ON_ values and finally "everything else".

Reply:
Sure - done.